crates/runtime/src/opentelemetry.rs

```diff
@@ -254,108 +353,170 @@ macro_rules! append_attribute {
+                                *double_value,
+                                Float64Builder,
+                                DataType::Float64,
+                                metric
+                            );
+                        }
+                        any_value::Value::BytesValue(bytes_value) => {
+                            append_attribute!(
+                                columns,
+                                fields,
+                                attribute.key,
+                                bytes_value,
+                                BinaryBuilder,
+                                DataType::Binary,
+                                metric
+                            );
+                        }
+                        // TODO: Support List and Map attribute types
+                        _ => {
+                            tracing::warn!(
+                                "Metric {metric} has attribute {key_str} with unsupported type, appending null for attribute if possible"
+                            );
+                            append_null(&mut fields, &mut columns, key_str);
+                        }
+                    },
+                    None => {
+                        tracing::warn!(
+                            "Metric {metric} has attribute {key_str} with no value, appending null for attribute if possible"
                         );
                         append_null(&mut fields, &mut columns, key_str);
                     }
                 },
                 None => {
+                    tracing::warn!(
+                        "Metric {metric} has attribute {key_str} with no value, appending null for attribute if possible"
+                    );
                     append_null(&mut fields, &mut columns, key_str);
                 }
-            },
-            None => {
-                append_null(&mut fields, &mut columns, key_str);
+            };
+        }
+
+        // If an attribute previously existed but is missing from this metric, append a null value.
+        let mut needs_null = Vec::new();
+        for (column_name, column_values) in columns.as_slice() {
+            if column_values.len() < i + 1 {
+                needs_null.push(column_name.clone());
             }
-        };
+        }
+        for column_name in needs_null {
+            append_null(&mut fields, &mut columns, column_name.as_str());
+        }
     }
 
     (fields, columns)
 }
 
+fn initialize_attribute_schema(
+    fields: &mut IndexMap<String, Arc<Field>>,
+    columns: &mut IndexMap<String, Box<dyn ArrayBuilder>>,
+    existing_schema: &Option<Schema>,
+) {
+    if let Some(s) = existing_schema {
+        for field in s.fields() {
+            // Skip value and time fields sbecause they are not attributes and are already handled.
+            if field.name() == VALUE_COLUMN_NAME
+                || field.name() == TIME_UNIX_NANO_COLUMN_NAME
```

Comment:
Is this column part of the OTel spec? Or something that telegraf is sending us?

Reply:
This is part of the OTel spec: https://github.com/open-telemetry/opentelemetry-proto/blob/4ca4f0335c63cda7ab31ea7ed70d6553aee14dce/opentelemetry/proto/metrics/v1/metrics.proto#L335